### PR TITLE
chore(profiling): refactor fetch libdatadog

### PR DESCRIPTION
**Pull request by sanchda**
This patch converts fetch libdatadog to a pure cmake implementation and updates the way the `dd_wrapper` is called so it should only be built once.

In current-main with an empty `build` directory:
```
 Performance counter stats for './build_standalone.sh -- -- --':

         63,864.14 msec task-clock:u              #    1.005 CPUs utilized
                 0      context-switches:u        #    0.000 /sec
                 0      cpu-migrations:u          #    0.000 /sec
         2,853,585      page-faults:u             #   44.682 K/sec
   <not supported>      cycles:u
   <not supported>      instructions:u
   <not supported>      branches:u
   <not supported>      branch-misses:u

      63.534837508 seconds time elapsed

      53.238101000 seconds user
      10.718485000 seconds sys
```

After applying this patch:
```
 Performance counter stats for './build_standalone.sh -- -- --':

         33,262.53 msec task-clock:u              #    0.985 CPUs utilized
                 0      context-switches:u        #    0.000 /sec
                 0      cpu-migrations:u          #    0.000 /sec
         1,638,895      page-faults:u             #   49.272 K/sec
   <not supported>      cycles:u
   <not supported>      instructions:u
   <not supported>      branches:u
   <not supported>      branch-misses:u

      33.757125924 seconds time elapsed

      27.434846000 seconds user
       5.887714000 seconds sys
```

So--a small difference, but a good difference.

## Checklist
- [X] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by nsrip-dd — APPROVED** [review]
I think you mentioned this before, but I assume this is so we can eventually set things up for our CMake build to only grab libdatadog once?

**Comment by sanchda** [comment]
> I think you mentioned this before, but I assume this is so we can eventually set things up for our CMake build to only grab libdatadog once?

Yeah, that's my hope.

File: ddtrace/internal/datadog/profiling/build_standalone.sh [hunk 1]
```diff
@@ -333,7 +333,9 @@ add_target() {
       targets+=("crashtracker")
       ;;
     dd_wrapper)
-      # We always build dd_wrapper, so no need to add it to the list
+      # `dd_wrapper` is a depencency of other targets, but the overall structure is weird when it's given explicitly
```
Comment (taegyunkim) [inline ddtrace/internal/datadog/profiling/build_standalone.sh:336]:
```suggestion
      # `dd_wrapper` is a dependency of other targets, but the overall structure is weird when it's given explicitly
```

**Comment by sanchda** [comment]
> > I think you mentioned this before, but I assume this is so we can eventually set things up for our CMake build to only grab libdatadog once?
> 
> Yeah, that's my hope.

Sorry @nsrip-dd , I accidentally made this PR actually do that thing, but also only build `dd_wrapper` once.  I meant to do those in followups.

**Comment by taegyunkim** [comment]
> > > I think you mentioned this before, but I assume this is so we can eventually set things up for our CMake build to only grab libdatadog once?
> > 
> > 
> > Yeah, that's my hope.
> 
> Sorry @nsrip-dd , I accidentally made this PR actually do that thing, but also only build `dd_wrapper` once. I meant to do those in followups.

Happy to see this change. This also saves some time to run sanitized native tests. As it used to run dd_wrapper test and then ran stack_v2 test.

**Review by taegyunkim — APPROVED** [review]

**Pull request merged**